### PR TITLE
[macOS SDK] Add Native Auth E2E and Integration tests for macOS

### PR DESCRIPTION
**Pull request by spetrescu84**
## Proposed changes

Describe what this PR is trying to do.
- Add E2E tests target for macOS
- Add Integration tests target for macOS
- Add E2E tests to be run as part of Automation validation (nightly and before release)

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [x] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: azure_pipelines/pr-validation.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
       testRunTitle: 'Test Run - $(target)'
 
 - job: e2e_test_native_auth
-  displayName: 'Run MSAL E2E tests for native auth'
+  displayName: 'Run MSAL E2E tests for iOS native auth'
```
Comment (nilo-ms) [inline azure_pipelines/pr-validation.yml:88]: Did we decide to run mac E2E tests also during PR validation? I thought we wanted to run those tests nightly and before the release (only in automation)

File: azure_pipelines/templates/tests-with-conf-file-mac.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+parameters:
```
Comment (nilo-ms) [inline azure_pipelines/templates/tests-with-conf-file-mac.yml:1]: Why do we need a new template? can we use the existing one?

File: azure_pipelines/templates/tests-with-conf-file.yml [hunk 3]
```diff
@@ -39,7 +39,7 @@ steps:
     inputs:
       targetType: 'inline'
       script: |
-        mv azure_pipelines/scripts/conf.json MSAL/test/automation/conf.json
+        cp azure_pipelines/scripts/conf.json MSAL/test/automation/conf.json
```
Comment (nilo-ms) [inline azure_pipelines/templates/tests-with-conf-file.yml:42]: is this change required? Why are we copying the conf.json file instead of moving it?

Comment (spetrescu84) [reply azure_pipelines/templates/tests-with-conf-file.yml]: It's not, I thought the conf file was getting downloaded on the main pipeline but it was actually not. Removed

File: azure_pipelines/pr-validation.yml (hunk 1 above)
Comment (spetrescu84) [reply azure_pipelines/pr-validation.yml]: I have now removed them as they also fail in the other case

File: MSAL/MSAL.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1059,7 +1091,7 @@
 		DECE0F292BE3EB010036738C /* MSALNativeAuthInternalError.swift in Sources */ = {isa = PBXBuildFile; fileRef = E206FC5E296D65DE00AF4400 /* MSALNativeAuthInternalError.swift */; };
 		DECE0F312BE3EB0B0036738C /* MSALNativeAuthCacheAccessor.swift in Sources */ = {isa = PBXBuildFile; fileRef = 289E156C2948EB8A006104D9 /* MSALNativeAuthCacheAccessor.swift */; };
 		DECE0F322BE3EB0D0036738C /* MSALNativeAuthCacheInterface.swift in Sources */ = {isa = PBXBuildFile; fileRef = 289E15582948E601006104D9 /* MSALNativeAuthCacheInterface.swift */; };
-		DECE0F332BE3EB0F0036738C /* MSALNativeAuthTokens.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2884855B295DAFD400516492 /* MSALNativeAuthTokens.swift */; };
+		DECE0F332BE3EB0F0036738C /* (null) in Sources */ = {isa = PBXBuildFile; };
```
Comment (diegojerezba) [inline MSAL/MSAL.xcodeproj/project.pbxproj:1094]: Is this `(null)` expected to be here? Same in line 6549

File: MSAL/MSAL.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -5263,8 +5389,6 @@
 					};
 					1E614BD922558D8300EBF62F = {
 						CreatedOnToolsVersion = 10.1;
-						DevelopmentTeam = UBF8T346G9;
```
Comment (diegojerezba) [inline MSAL/MSAL.xcodeproj/project.pbxproj:5266]: Is this change expected?

File: azure_pipelines/templates/tests-with-conf-file-mac.yml (hunk 2 above)
Comment (diegojerezba) [reply azure_pipelines/templates/tests-with-conf-file-mac.yml:1]: I agree with @nilo-ms , it would better to avoid creating a similar script if possible.

You can see an example of an script with arguments in the `spm-integration-test.sh`. It is called from `pr-validation.yml`, and the name of the branch is passed as an argument.

File: MSAL.xcworkspace/contents.xcworkspacedata [hunk 6]
```diff
@@ -4,7 +4,4 @@
    <FileRef
       location = "group:MSAL/MSAL.xcodeproj">
    </FileRef>
-   <FileRef
-      location = "group:MSAL/MSAL iOS Native Auth E2E Tests.xctestplan">
```
Comment (diegojerezba) [inline MSAL.xcworkspace/contents.xcworkspacedata:8]: Do you know why this shows as removed? I see that the `MSAL iOS Native Auth E2E Tests.xctestplan` is still in the project, so it all should be good, but I'm not sure why this change appears here.

File: azure_pipelines/templates/tests-with-conf-file-mac.yml (hunk 2 above)
Comment (spetrescu84) [reply azure_pipelines/templates/tests-with-conf-file-mac.yml:1]: Indeed, the problem here is that the script itself has more lines, it's not about arguments but rather these extra 3 lines:
```
macos_version=$(sw_vers -productVersion)
echo "Mac Version $macos_version"      
full_path="build/Build/Products/${{ parameters.build }}_macosx$macos_version-x86_64.xctestrun"
```
I don't see a way to have them in the same script without parametrising the script itself

File: MSAL/MSAL.xcodeproj/project.pbxproj (hunk 4 above)
Comment (spetrescu84) [reply MSAL/MSAL.xcodeproj/project.pbxproj]: Removed and all tests look still ok

File: azure_pipelines/pr-validation.yml (hunk 1 above)
Comment (nilo-ms) [reply azure_pipelines/pr-validation.yml]: Are those tests run in parallel with the iOS E2E tests?

File: MSAL/MSAL.xcodeproj/project.pbxproj (hunk 5 above)
Comment (spetrescu84) [reply MSAL/MSAL.xcodeproj/project.pbxproj:5277]: Reverted

File: azure_pipelines/templates/tests-with-conf-file-mac.yml (hunk 2 above)
Comment (nilo-ms) [reply azure_pipelines/templates/tests-with-conf-file-mac.yml:1]: Can you add the xcodebuild command in a parameter and use the iOS as default parameter value?

File: azure_pipelines/pr-validation.yml (hunk 1 above)
Comment (spetrescu84) [reply azure_pipelines/pr-validation.yml]: They would be run in parallel on a different machine indeed

File: azure_pipelines/templates/tests-with-conf-file-mac.yml (hunk 2 above)
Comment (spetrescu84) [reply azure_pipelines/templates/tests-with-conf-file-mac.yml]: Parametrised the `full_path`, `sdk`and `destination` parameters with defaults for MSAL UI tests

File: MSAL/test/integration/native_auth/end_to_end/sign_out/MSALNativeAuthSignOutEndToEndTests.swift [hunk 7]
```diff
@@ -28,6 +28,10 @@ import XCTest
 final class MSALNativeAuthSignOutEndToEndTests: MSALNativeAuthEndToEndPasswordTestCase {
     // Hero Scenario 2.4.1. Sign out – Local sign out from app on device (no SSO)
     func test_signOutAfterSignInPasswordSuccess() async throws {
+        // TOOD: This will be re-enabled as part of another PBI
```
Comment (nilo-ms) [inline MSAL/test/integration/native_auth/end_to_end/sign_out/MSALNativeAuthSignOutEndToEndTests.swift:31]: nit- typo TOOD

Comment (spetrescu84) [reply MSAL/test/integration/native_auth/end_to_end/sign_out/MSALNativeAuthSignOutEndToEndTests.swift]: Sorted

**Review by diegojerezba — APPROVED** [review]

**Review by ameyapat — APPROVED** [review]
